### DISCUSSION TITLE
Add Deprecate Action capability

### DISCUSSION THREAD

**Discussion by em-c-rod** [community/community#12564] [Actions]
When deprecating an action, there is an option to deprecate the inputs with [inputs.<input_id>.deprecationMessage](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddeprecationmessage), but there is no option to deprecate the entire action and notify the user of it. 

Below are the results of an example workflow output that could be misleading to the user because it states “Input ‘file’ has been deprecated with message: Our deprecation message for the entire action”. In reality, it is the action, not the input, that is being deprecated.

![image](https://user-images.githubusercontent.com/32680215/157310269-e6f2ec3f-7a3c-440c-95aa-5310bf2763f3.png)

Instead, it would be ideal to notify the user upon calling the action that the action itself is deprecated.

**Comment by aaronlippold** [comment]
Very much needed for positive project and service managment

**Comment by ethomson** [comment]
Thanks for the feedback @em-c-rod - I can see how this would be useful. I'll make sure that this is on the backlog.

**Comment by kerbou** [comment]
@ethomson Any news on this? As a platform developer on a platform team writing actions used by multiple product teams in various combinations, we would love a better developer experience when communicating to teams which actions they should stop using going forward.

**Reply by elrayle** [reply]
And ideally, like many deprecation messages allow/encourage, what should be done instead (e.g. a replacement action to use).  Any movement on this?

**Reply by andyloughran** [reply]
I'm keen to know how people get around this issue - this looks like exactly what I think I want for very similar reasons.
